### DISCUSSION TITLE
WPF LiveCharts

### DISCUSSION THREAD

**Discussion by Zifanghenmang** [BYJRK/DotNet-Discussions#124] [WPF 开发]
大佬，讲下WPF中图表控件或者其他好用的包，展示数据的

**Comment by CodingOctocat** [comment]
比较好用的应该是（我也没用过-_-）：https://oxyplot.github.io/

**Comment by pchuan98** [comment]
那可太多了，

首先是微软当初发布WPF的时候就发布过一个图表开源库，大致名字不太记得了，但是可以用结尾的方法查到，然后第三方最多人用的有livechart和livecharts2，下面是我做三维项目的简单调研做的一个图表，仅供参考

| 名称           | 简单控件 |   2D   |   3D   | GPU渲染 |   支持程度   |   价格   |                                       网站                                       |                  其他                   |
| -------------- |:--------:|:------:|:------:|:-------:|:------------:|:--------:|:--------------------------------------------------------------------------------:|:---------------------------------------:|
| ActiViz        |  不支持  |  支持  |  支持  |  未知   | 提供解决方案 |   800€   |                [Activiz](https://www.kitware.eu/activiz-training)                |                    -                    |
| Xceed          |   支持   |  支持  |  支持  |  未知   |   开发文档   |    -     | [Xceed Software Inc](https://xceed.com/en/our-products/products/category/ui/wpf) |             有很多工具类库              |
| Halcon         |  不支持  |  支持  | 不支持 |  未知   |   开发文档   |    -     |               [MVTec Software](https://www.mvtec.com/cn/products)                |         太重了，不适合直接上手          |
| Actipro        |   支持   |  支持  | 不支持 |  未知   |   开发文档   |  699＄   |               [Actipro Software](https://www.actiprosoftware.com/)               |                    -                    |
| DevExpress     |   支持   |  支持  |  支持  |  未知   |   开发文档   | 899.99＄ |   [WPF Charts](https://www.devexpress.com/products/net/controls/wpf/charting/)   | 非常适合，但是不知道是否支持GPU加速方案 |
| Helix Toolkits |  不支持  | 不支持 |  支持  |  DirectX9   |   开发文档   |   free   | [helix-toolkit](https://github.com/helix-toolkit/helix-toolkit)  |    方案单一但是开源                                     |

其实最简单、最合理的办法就是直接去github搜索，关键字chart，然后排序most starts，限制语言C#，现在大多数图表库都是多平台的，很方便。

**Reply by pchuan98** [reply]
就性能而言，找个能对接opengl的一般来说都挺快的

**Comment by LiuChao19940201** [comment]
可是试试这个：https://scottplot.net/
这个表格控件还可以，不卡

**Comment by lqy16** [comment]
之前用的是livecharts，但是作者现在在写2，还不太完善
最新的项目用的是scottplot，缺点是不支持绑定式的开发（只支持事件式），所以可能需要自己定制控件
